src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -858,6 +860,7 @@ else if (shouldBootstrap())
             // (we won't be part of the storage ring though until we add a counterId to our state, below.)
             // Seed the host ID-to-endpoint map with our own ID.
             UUID localHostId = SystemKeyspace.getLocalHostId();
+            VolumeIntegrityCheck.of(localHostId).execute();
```

Comment:
Not sure that logically, this should be part of the `prepareToJoin` function. We should move it up to `initServer`

Reply:
At this point, we know for sure that the host ID exists locally. One concern for calling the check earlier is that we alter Cassandra's behavior, which may lead to unintended consequences.

Reply:
Ah because `getLocalHostId` is what actually creates the local host id if it does not exists. Oki, sg